server/src/main/java/org/opensearch/action/search/SearchTask.java

```diff
@@ -80,7 +80,11 @@ public SearchTask(
 
     @Override
     public final String getDescription() {
-        return descriptionSupplier.get();
+        try {
+            return descriptionSupplier.get();
+        } catch (UnsupportedOperationException e) {
```

Comment:
UnsupportedOperationException is broad category of error and we shouldn't catch it and swallow. This may help in surfacing more issues like this in future.

do you want to rethrow this as wrapped OpenSearchException instead of swallowing the exception

Reply:
This won't fix the original problem where with the trace enabled the user-facing behavior is different.

Reply:
Ideally this should be caught in the validation when it is creating the search request upfront whether a shape is supported or not in ```AbstractGeometryQueryBuilder``` instead of generating the request, create SearchTask for it and then fail during execution. Failing fast is better in case it is not supported at all.

Reply:
@dblock I think there is a way to solve in a way to make the behaviour acceptable:

- `SearchSourceBuilder`:

```
    public String toString(Params params) {
        try {
            return XContentHelper.toXContent(this, MediaTypeRegistry.JSON, params, true).utf8ToString();
        } catch (IOException e) {
            throw new OpenSearchException(e);
        } catch (UnsupportedOperationException ex) {
            throw new OpenSearchException(ex);
        }
    }
```

- `SearchRequest`
```
    public final String buildDescription() {
        StringBuilder sb = new StringBuilder();
        sb.append("indices[");
        Strings.arrayToDelimitedString(indices, ",", sb);
        sb.append("], ");
        sb.append("search_type[").append(searchType).append("], ");
        if (scroll != null) {
            sb.append("scroll[").append(scroll.keepAlive()).append("], ");
        }
        if (source != null) {
            try {
                sb.append("source[").append(source.toString(FORMAT_PARAMS)).append("]");
            } catch (final OpenSearchException ex) {
                sb.append("source[").append("n/a").append("]");
            }
        } else {
            sb.append("source[]");
        }
        return sb.toString();
    }

```

Reply:
On a different note, it would be good to repro this error first. As per documentation there is no direct way to pass LinearRing as shape type - https://opensearch.org/docs/latest/field-types/supported-field-types/geo-shape/
It should fail in parsing, early on.

Only Polygon type internally uses this. 
https://github.com/opensearch-project/OpenSearch/blob/e673b6194fab6f672b62dfdaf8f082bc8dec403f/server/src/main/java/org/opensearch/common/geo/builders/PolygonBuilder.java#L72

From the documentation
```
          Per GeoJSON spec (http://geojson.org/geojson-spec.html#linestring)
          A LinearRing is closed LineString with 4 or more positions. The first and last positions
          are equivalent (they represent equivalent points). Though a LinearRing is not explicitly
          represented as a GeoJSON geometry type, it is referred to in the Polygon geometry type definition.
```

Just check, this problem does exist today.

Reply:
@shwetathareja correct, one the ways to fail earlier is to add checks in `GeoShapeQueryBuilder` / `AbstractGeometryQueryBuilder`, the issue though is that this class deals with `Supplier<Geometry>` (the resolution is delayed later).